src/evm/engine_db/simulation_db.rs

```diff
@@ -242,22 +230,26 @@ where
     ///
     /// # Returns
     ///
-    /// Returns a `Result` containing the value from storage at the specified index as an `rU256`,
+    /// Returns a `Result` containing the value from storage at the specified index as an `U256`,
     /// or an error of type `SimulationDB<M>::Error` if the query fails.
     pub fn query_storage(
         &self,
         address: Address,
-        index: rU256,
-    ) -> Result<rU256, <SimulationDB<M> as DatabaseRef>::Error> {
-        let index_h256 = H256::from(index.to_be_bytes());
+        index: U256,
+    ) -> Result<StorageValue, <SimulationDB<P> as DatabaseRef>::Error> {
         let fut = async {
-            let address = H160::from(**address);
-            let storage = self
-                .client
-                .get_storage_at(address, index_h256, self.block.map(|v| v.into()))
-                .await
-                .unwrap();
-            rU256::from_be_bytes(storage.to_fixed_bytes())
+            if let Some(ref block) = self.block {
+                self.client
+                    .get_storage_at(address, index)
+                    .number(block.number)
+                    .await
+                    .unwrap()
+            } else {
+                self.client
+                    .get_storage_at(address, index)
+                    .await
+                    .unwrap()
+            }
```

Comment:
I am a bit confused about the implications of this change, why are we still using `block_on`? Is that necessary now with an async provider? Or will we just address this in a future PR?

```
    fn block_on<F: core::future::Future>(&self, f: F) -> F::Output {
        // If we get here and have to block the current thread, we really
        // messed up indexing / filling the storage. In that case this will save us
        // at the price of a very high time penalty.
        match &self.runtime {
            Some(runtime) => runtime.block_on(f),
            None => futures::executor::block_on(f),
        }
    }

Reply:
yeah I think we confused ourselves. ethers Middleware was async as well. (See [here](https://docs.rs/ethers-providers/2.0.14/src/ethers_providers/middleware.rs.html#147)). it's the DatabaseRef that isn't async. so we need to wrap the asyncness in the block_on to be sync.
when the async databaseref is release we can change our code to use it and remove this block_on!

Reply:
ahhhhh ok nice